nntrainer/layers/time_dist.h

```diff
@@ -97,6 +97,11 @@ class TimeDistLayer : public Layer {
    */
   std::shared_ptr<Layer> &getDistLayer() { return dist_layer; };
 
+  /**
+   * @copydoc Layer::requireLabe()
+   */
+  bool requireLabel() const override { return dist_layer->requireLabel(); }
```

Comment:
can you consider not adding this here, but rather move this to LayerNode and use getLayer()->requireLabel().
This will cover the dist layer case automatically.
The purpose is to keep dist layer as light as possible, otherwise management becomes tough.

Reply:
You mean sthlike below?

```cpp
LayerNode::requireLabe() {
 if (type==timedist_layer) {
     return layer->getLayer()->requireLabel()
 }
}
```

If it is, or not, Could you ping me if you've got time to discuss offline?

Reply:
no, the below code precisely:
```cpp
const bool LayerNode::requireLabe() const {
     return getLayer()->requireLabel();
}
```

Reply:
Ah, yeah this has to be done, let me do it in follow up PR

Reply:
then make sure to remove the corresponding function from time dist layer.

Reply:
Yup